CMakeLists.txt

```diff
@@ -0,0 +1,345 @@
+file(GLOB_RECURSE FNT_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/data/fonts/ ${PROJECT_SOURCE_DIR}/data/fonts/*.ttf)
+set(AUS_FILES
+	audio.json) # Not using GLOB here since its just a single file, and for audio. There seems to be some other json files in the data directories too, but they are used for bitmask stuff?
+file(GLOB_RECURSE SND_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/data/sounds/ ${PROJECT_SOURCE_DIR}/data/sounds/*.wav)
+
+# Source files
+file(GLOB_RECURSE IWRAM_C_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.iwram.c)
+file(GLOB_RECURSE IWRAM_CPP_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.iwram.cpp)
+file(GLOB_RECURSE GAME_C_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.c)
+list(FILTER GAME_C_FILES EXCLUDE REGEX ".*iwram.c$")
+file(GLOB_RECURSE GAME_CPP_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.cpp)
+list(FILTER GAME_CPP_FILES EXCLUDE REGEX ".*iwram.cpp$")
+file(GLOB_RECURSE GAME_ASM_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.s)
+
+# Build Functions
+function(build_resource)
+	# Set Variables to make this easier
+	# i.e SPRITE
+	set(RESOURCE_NAME ${ARGV0})
+	# i.e SPRITES
+	set(PLURAL ${RESOURCE_NAME}S)
+	# passed in dependencies, tool or the previous object
+	set(TARGET_DEPENDENCIES ${ARGV1})
+	# Set helped variables based on what resource we are exporting and building
+	if(${RESOURCE_NAME} MATCHES "SPRITE")
+		set(TOOL_NAME sprite-export)
+		set(FOLDER_NAME sprites/)
+		set(OUT1 .s)
+		set(OUT2 .hpp)
+	elseif(${RESOURCE_NAME} MATCHES "BACKGROUND")
+		set(TOOL_NAME background-export)
+		set(FOLDER_NAME backgrounds/)
+		set(OUT1 .s)
+		set(OUT2 .hpp)
+	elseif(${RESOURCE_NAME} MATCHES "FONT")
+		set(TOOL_NAME font-export)
+		set(FOLDER_NAME fonts/)
+		set(OUT1 .s)
+		set(OUT2 .hpp)
+	elseif(${RESOURCE_NAME} MATCHES "SETTING")
+		set(TOOL_NAME audio-export-settings)
+		set(FOLDER_NAME "")
+		set(OUT1 -settings.hpp)
+		set(OUT2 -settings.json)
+	elseif(${RESOURCE_NAME} MATCHES "SOUND")
+		set(TOOL_NAME sound-export)
+		set(FOLDER_NAME sounds/)
+		set(OUT1 .s)
+		set(OUT2 .hpp)
+	endif()
+	set(RESOURCE_FILES ${ARGV2})
+	set(IN2 ${ARGV3})
+
+	#make a directory in build folder for the output files
+	file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/data/${FOLDER_NAME})
+
+	set(OBJECT_LIST_NAME ${PLURAL}_OBJECT_LIST)
+	set(OUT1_LIST_NAME ${PLURAL}_OUT1_LIST)
+	set(OUT2_LIST_NAME ${PLURAL}_OUT2_LIST)
+
+	foreach(FILE ${RESOURCE_FILES})
+		get_filename_component(file_dir "${FILE}" PATH)
+		string(REPLACE "." ";." file ${FILE})
+		list(GET file 0 file_name)
+		list(GET file 1 file_type)
+
+		set(OUTPUT_FILE_1 ${CMAKE_BINARY_DIR}/data/${FOLDER_NAME}${file_name}${OUT1})
+		set(OUTPUT_FILE_2 ${CMAKE_BINARY_DIR}/data/${FOLDER_NAME}${file_name}${OUT2})
```

Comment:
Here, instead of having separate `OUTPUT_FILE_n` variables, you can use a list transformer, since you're using CMake 3.12+ anyway.

    list(TRANSFORM OUT PREPEND ${CMAKE_BINARY_DIR}/data/${FOLDER_NAME}${file_name} OUTPUT_VARIABLE OUTPUT_FILES)